walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -48,30 +49,30 @@
 
     Collection<TimeRange> possibleTimes = new ArrayList<>();
     if (attendees.isEmpty()) {
-      // If there is no mandatory attendee, only consider availabilities of optional attendees.
+      // Case: No mandatory attendee, only consider availabilities of optional attendees.
 
-      // {@code eventTimes} is the list of current events' time ranges that need to be avoided when scheduling.
+      // A list of current events' time ranges that need to be avoided when scheduling.
       List<TimeRange> eventTimes = new ArrayList<>();
-      for (List<TimeRange> timeList : optionalAttendeesEventTimes.values()) {
+      for (List<TimeRange> timeList : this.optionalAttendeesEventTimes.values()) {
         eventTimes.addAll(timeList);
       }
       Collections.sort(eventTimes, TimeRange.ORDER_BY_START);
       possibleTimes = findTimeRangeGaps(eventTimes, duration);
     } else {
-      // If there are mandatory attendees, consider optional attendees who can possibly attend and ignore those who can't. 
+      // Case: There are mandatory attendees, consider optional attendees who can possibly attend and ignore those who can't. 
 
-      List<TimeRange> eventTimes = new ArrayList<>(attendeesEventTimes);
+      List<TimeRange> eventTimes = new ArrayList<>(this.attendeesEventTimes);
 
       // Find the time slot(s) that maximize the number of optional attendees who can attend.
       // Count down from the maximum number of optional attendees and find all combinations of that number of attendees.
       // Once we find a combination where scheduling is possible, we've found the maximal group of attendees.
       for (int num = optionalAttendees.size(); num > 0; num--) {
-        String[] arr = optionalAttendees.toArray(new String[optionalAttendees.size()]);
-        getAllCombinations(arr, arr.length, num);
+        String[] optionalAttendeesArr = optionalAttendees.toArray(new String[optionalAttendees.size()]);
```

Comment:
`optionalAttendeesArray`

Reply:
Could we change this to use an [ImmutableList](https://guava.dev/releases/23.0/api/docs/com/google/common/collect/ImmutableList.html) instead?

Reply:
I've never used an ImmutableList, but sure